data-access/src/object_store/local.rs

```diff
@@ -50,17 +50,63 @@ impl ObjectStore for LocalFileSystem {
 
     async fn list_dir(
         &self,
-        _prefix: &str,
-        _delimiter: Option<String>,
+        prefix: &str,
+        delimiter: Option<String>,
     ) -> Result<ListEntryStream> {
-        todo!()
+        if let Some(d) = delimiter {
+            if d != "/" && d != "\\" {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::InvalidInput,
+                    format!("delimiter not supported on local filesystem: {}", d),
+                ));
+            }
+            let mut entry_stream = tokio::fs::read_dir(prefix).await?;
+
+            let list_entries = stream::poll_fn(move |cx| {
```

Comment:
I think you may be able to use https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html here to `collect` the stream

Maybe something like (untested):

```rust
let list_entries = entry_stream
  .collect::<Result<Vec<_>>>()
  .await?
  .then(...);
```

Reply:
I could, but was attempting to not collect, just in case the caller didn't want to collect all results in memory. It's definitely more verbose as a result though.